igwn_auth_utils/requests.py

```diff
@@ -0,0 +1,360 @@
+# -*- coding: utf-8 -*-
+# Copyright 2021 Cardiff University
+# Distributed under the terms of the BSD-3-Clause license
+
+"""Python Requests interface with IGWN authentication
+
+This is heavily inspired by Leo Singer's excellent `requests-gracedb` package.
+"""
+
+__author__ = "Duncan Macleod <duncan.macleod@ligo.org>"
+__credits__ = "Leo Singer <leo.singer@ligo.org>"
+
+from os import getenv
+from netrc import netrc
+from urllib.parse import urlparse
+
+import requests
+
+from . import (
+    IgwnAuthError,
+    find_x509_credentials,
+    find_scitoken,
+    scitoken_authorization_header,
+)
+
+_auth_session_parameters = """
+    Discovery/configuration of authorisation/authentication methods
+    is attempted in the following order:
+
+    1.  if ``force_noauth=True`` is given, no auth is configured;
+
+    3.  if a bearer token is provided via the ``token`` keyword argument,
+        then use that;
+
+    3.  if an X.509 credential path is provided via the ``cert`` keyword
+        argument, then use that;
```

Comment:
Also document order in which environment variables, /tmp, and .globus dirs are checked.

Reply:
This is referenced below:

```
    6.  look for an X.509 credential using
        :func:`igwn_auth_utils.find_x509_credential`
```

the sphinx docs will include a link to the documentation of the relevant function, which lists the ordering. I'm hesitant to duplicate that documentation here for fear that the two copies will get out of sync.